test/test.js

```diff
@@ -1339,6 +1339,44 @@ describe('Page', function() {
     }));
   });
 
+  describe('Request.respond', function() {
+    it('should work', SX(async function() {
+      await page.setRequestInterceptionEnabled(true);
+      page.on('request', request => {
+        request.respond({
+          status: 201,
+          headers: {
+            foo: 'bar'
+          },
+          body: Buffer.from('Yo, page!', 'utf8')
+        });
+      });
+      const response = await page.goto(EMPTY_PAGE);
+      expect(response.status).toBe(201);
+      expect(response.headers.foo).toBe('bar');
+      expect(await page.evaluate(() => document.body.textContent)).toBe('Yo, page!');
+    }));
+    it('should allow mocking binary responses', SX(async function() {
+      await page.setRequestInterceptionEnabled(true);
+      const imageBuffer = fs.readFileSync(path.join(__dirname, 'assets', 'pptr.png'));
+      page.on('request', request => {
+        request.respond({
+          contentType: 'image/png',
+          body: imageBuffer
+        });
+      });
+      await page.goto(EMPTY_PAGE);
```

Comment:
I am confused what this test is doing. You are loading the image, and intercepting it with the same exact image? And won't the initial navigation also be overridden as the image?

I would expect there to be two images. And then bad.png is overridden with good.png.

Reply:
This is a funny one :) Fixed.